Rubberduck.Refactorings/ReorderParameters/IReorderParametersPresenter.cs

```diff
@@ -1,7 +1,5 @@
 ﻿namespace Rubberduck.Refactorings.ReorderParameters
 {
-    public interface IReorderParametersPresenter
-    {
-        ReorderParametersModel Show();
-    }
+    public interface IReorderParametersPresenter : IRefactoringPresenter<ReorderParametersModel>
```

Comment:
I'm not sure if there's any point in having a empty interface just to implement a closed generic interface. Castle Windsor can be updated to correctly map the closed generic type without a closed interface, similar to what is done in `RegisterRefactoringDialogs` method of the IOC installer.

However,  `RenamePresenter` does in fact implement 2 other members - I'm not sure if we really need them but if we do, then it may need to be resolved before using the generic selector.

Reply:
I think the two additional methods for rename are redundant now. However, I am not fully sure whether the dialog binds to them.

For this PR, I avoided as much as possible to touch the view infrastructure.

Reply:
Understood - that can be addressed in a separate PR.